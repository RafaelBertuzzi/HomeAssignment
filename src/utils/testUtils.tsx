import { ReactNode } from 'react';
import renderer from 'react-test-renderer';

import { LayoutProvider } from '@components/LayoutProvider';
import { render } from '@testing-library/react-native';

export const rendererWithProviders = (children: ReactNode) => {
  return renderer.create(<LayoutProvider>{children}</LayoutProvider>).toJSON();
};

export const renderWithProviders = (children: ReactNode) => {
  return render(<LayoutProvider>{children}</LayoutProvider>).toJSON();
};
