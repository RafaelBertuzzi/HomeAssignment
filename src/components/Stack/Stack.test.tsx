import { Text } from 'react-native';

import { cleanup } from '@testing-library/react-native';
import { rendererWithProviders } from '@utils/testUtils';

import { Stack } from '.';

const texts = ['Stack 1', 'Stack 2'].map((text) => (
  <Text key={text}>{text}</Text>
));

describe('Component: Stack', () => {
  afterEach(cleanup);

  it('should render children vertical', () => {
    const json = rendererWithProviders(<Stack>{texts}</Stack>);
    expect(json).toMatchSnapshot();
  });

  it('should render children horizontal', () => {
    const json = rendererWithProviders(
      <Stack direction="column">{texts}</Stack>,
    );

    expect(json).toMatchSnapshot();
  });

  it('should render space', () => {
    const json = rendererWithProviders(<Stack space="md">{texts}</Stack>);
    expect(json).toMatchSnapshot();
  });
});
