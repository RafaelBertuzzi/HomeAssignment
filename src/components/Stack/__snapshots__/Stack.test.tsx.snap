// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Stack should render children horizontal 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    dataSet={{}}
    style={
      {
        "flexDirection": "column",
      }
    }
  >
    <Text>
      Stack 1
    </Text>
    <View
      dataSet={{}}
      style={{}}
    />
    <Text>
      Stack 2
    </Text>
  </View>
</RNCSafeAreaProvider>
`;

exports[`Component: Stack should render children vertical 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    dataSet={{}}
    style={{}}
  >
    <Text>
      Stack 1
    </Text>
    <View
      dataSet={{}}
      style={{}}
    />
    <Text>
      Stack 2
    </Text>
  </View>
</RNCSafeAreaProvider>
`;

exports[`Component: Stack should render space 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    dataSet={{}}
    style={{}}
  >
    <Text>
      Stack 1
    </Text>
    <View
      dataSet={{}}
      style={
        {
          "height": 16,
        }
      }
    />
    <Text>
      Stack 2
    </Text>
  </View>
</RNCSafeAreaProvider>
`;
